klusterletconfig/v1alpha1/types.go

```diff
@@ -58,6 +64,25 @@ type Registries struct {
 	Source string `json:"source"`
 }
 
+// KubeAPIServerProxyConfig describes the proxy settings for the connections to a kube-apiserver
+type KubeAPIServerProxyConfig struct {
+	// HTTPProxy is the URL of the proxy for HTTP requests
+	// +optional
+	HTTPProxy string `json:"httpProxy,omitempty"`
```

Comment:
can we set httpproxy and httpsproxy at the same time?

Reply:
No, only one of them can be picked up. Because there is only one field `proxy-url` in kubeconfig.
```
apiVersion: v1
clusters:
  - cluster:
      certificate-authority-data: LS0tLS...LS0tCg==
      server: https://api.server-foundation-sno-x4lcs.dev04.red-chesterfield.com:6443
      proxy-url: https://10.0.109.153:3129
    name: default-cluster
```

Reply:
If both are set, which one will be picked? I think we need to document it or we should have have only one field.

Reply:
It has been documented here: https://github.com/stolostron/cluster-lifecycle-api/pull/18/files#diff-8f274b725ecd26693611954469b6d04ef680e312cba3b0175fa4bd44e91e5f28R74